modules/statistics/php/stats_demographic.class.inc

```diff
@@ -163,65 +167,80 @@ class Stats_Demographic extends \NDB_Form
         foreach (\Utility::getProjectList() as $key => $value) {
             $projects[$key] = $value;
         }
-        $currentProject = null;
-        if ($_REQUEST['DemographicProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['DemographicProject']);
+        $demographicProject = $_REQUEST['DemographicProject'] ?? '';
+        $Param_Project      = '';
+
+        if (empty($demographicProject)) {
+            error_log('No demographic project specified in request');
+        } else {
+            $demographicProject = htmlspecialchars($demographicProject);
```

Comment:
should this be on line 170 ?

Reply:
Why? This way we only call `htmlspecialchars` if `$demographicProject` isn't empty. It isn't used between then.

Reply:
you are altering the behaviour. Before $currentProject was set to null unless you go in the ifstatement, now $demographic project is set to a value Which is not being passed through htmlspecialchars unless it goes through the else statement. there is a difference between `' '` and `null`

Reply:
ok